internal/contentcache/contentcache.go

```diff
@@ -44,10 +44,51 @@ type CacheObjectKey struct {
 type ContentCache struct {
 	debug      *log.Logger
 	tempDir    string
-	fileMap    map[CacheObjectKey]gcsx.TempFile
+	fileMap    map[CacheObjectKey]*CacheObject
 	mtimeClock timeutil.Clock
 }
 
+// Metadata store struct
+type CacheFileObjectMetadata struct {
```

Comment:
This needs a unit test

Reply:
I will make a separate PR to create some basic unit tests for content_cache.go

Reply:
The unit test for X should be put in the same PR where X is introduced.

A recommended fix for this PR is to create another PR2 to introduce CacheFileObjectMetadata and its test, and rebase this PR onto the PR2.